src/crimson/osd/pg_shard_manager.h

```diff
@@ -163,11 +194,11 @@ class PGShardManager {
       op->get_pgid());
 
     get_local_state().registry.remove_from_registry(*op);
-    return with_remote_shard_state(
-      core,
-      [op=std::move(op)](
```

Comment:
Although T::IRef is a boost::intrusive_ptr<T>, but looks it works like unique-ptr with `ceph_assert(op->use_count() == 1)`.

However, this still looks tricky to move the `op` to another core directly, it seems to result in allocating the `op` in one core and deallocate it in a different core, which should be wrong but I don't know why there is no visible failure.

Same applies to other unique-ptrs that may be owned by a different core, for example `MessageURef` should be a foreign-ptr.

Reply:
yes, that make me confused also.

Reply:
Ah, I see that seastar allocator can recognize and handle foreign free internally. So it is fine.